src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509AuthorityKeyIdentifierExtension.cs

```diff
@@ -0,0 +1,575 @@
+
+                writer.WriteIntegerUnsigned(serialNumber, new Asn1Tag(TagClass.ContextSpecific, 2));
+            }
+
+            return new X509AuthorityKeyIdentifierExtension(writer.Encode());
+        }
+
+        /// <summary>
+        ///   Creates an <see cref="X509AuthorityKeyIdentifierExtension"/> based on values
+        ///   from the provided certificate.
+        /// </summary>
+        /// <param name="certificate">
+        ///   The Certificate Authority (CA) certificate that will sign this extension.
+        /// </param>
+        /// <param name="includeKeyIdentifier">
+        ///   <see langword="true" /> to include the Subject Key Identifier value from the certificate
+        ///   as the key identifier value in this extension; otherwise, <see langword="false" />.
+        /// </param>
+        /// <param name="includeIssuerAndSerial">
+        ///   <see langword="true" /> to include the certificate's issuer name and serial number
+        ///   in this extension; otherwise, <see langword="false" />.
+        /// </param>
+        /// <returns>
+        ///   The configured extension.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="certificate"/> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <paramref name="includeKeyIdentifier" /> is <see langword="true" />, but
+        ///   <paramref name="certificate" /> does not contain a Subject Key Identifier extension.
+        /// </exception>
+        public static X509AuthorityKeyIdentifierExtension CreateFromCertificate(
+            X509Certificate2 certificate,
+            bool includeKeyIdentifier,
+            bool includeIssuerAndSerial)
+        {
+            ArgumentNullException.ThrowIfNull(certificate);
+
+            if (includeKeyIdentifier)
+            {
+                X509SubjectKeyIdentifierExtension? skid =
+                    (X509SubjectKeyIdentifierExtension?)certificate.Extensions[Oids.SubjectKeyIdentifier];
+
+                if (skid is null)
+                {
+                    throw new CryptographicException(SR.Cryptography_X509_AKID_NoSKID);
+                }
+
+                ReadOnlySpan<byte> skidBytes = skid.SubjectKeyIdentifierBytes.Span;
+
+                if (includeIssuerAndSerial)
+                {
+                    return Create(
+                        skidBytes,
+                        certificate.IssuerName,
+                        certificate.SerialNumberBytes.Span);
+                }
+
+                return CreateFromSubjectKeyIdentifier(skidBytes);
+            }
+            else if (includeIssuerAndSerial)
+            {
+                return CreateFromIssuerNameAndSerialNumber(
+                    certificate.IssuerName,
+                    certificate.SerialNumberBytes.Span);
+            }
+
+            Span<byte> emptyExtension = new byte[] { 0x30, 0x00 };
```

Comment:
```suggestion
            ReadOnlySpan<byte> emptyExtension = new byte[] { 0x30, 0x00 };
```

It will allocate otherwise.

Reply:
This was also marked resolved without a change?